src/clr-angular/forms/datepicker/date-input.spec.ts

```diff
@@ -0,0 +1,275 @@
+
+                it("sets the date on the IOService", () => {
+                    const date: Date = new Date();
+
+                    expect(dateIOService.date).toBeUndefined();
+
+                    context.clarityDirective.date = date;
+
+                    expect(dateIOService.date).not.toBeUndefined();
+                    expect(dateIOService.date.getDate()).toBe(date.getDate());
+                    expect(dateIOService.date.getMonth()).toBe(date.getMonth());
+                    expect(dateIOService.date.getFullYear()).toBe(date.getFullYear());
+
+                    const inValidDate: Date = new Date("Test");
+
+                    context.clarityDirective.date = inValidDate;
+                    expect(dateIOService.date).toBeNull();
+                });
+            });
+
+            describe("Host Bindings & Listeners", () => {
+                it("binds the placeholderText to the placeholder of the input", () => {
+                    expect(context.clarityElement.placeholder).toBe("MM/DD/YYYY");
+                });
+
+                it("binds the input type correctly", () => {
+                    expect(context.clarityElement.type).toBe("text");
+
+                    enabledService.fakeIsEnabled = false;
+                    context.detectChanges();
+
+                    expect(context.clarityElement.type).toBe("date");
+                });
+
+                it("listens to the input change events", () => {
+                    spyOn(context.clarityDirective, "onValueChange");
+
+                    const inputEl = context.fixture.debugElement.query(By.directive(ClrDateInput));
+                    inputEl.triggerEventHandler("change", inputEl);
+
+                    expect(context.clarityDirective.onValueChange).toHaveBeenCalled();
+                });
+            });
+        });
+
+        describe("Datepicker with ngModel", () => {
+            let fixture: ComponentFixture<TestComponentWithNgModel>;
+            let compiled: any;
+            let dateContainerDebugElement: DebugElement;
+            let dateInputDebugElement: DebugElement;
+
+            beforeEach(function() {
+                TestBed.configureTestingModule(
+                    {imports: [FormsModule, ClrFormsModule], declarations: [TestComponentWithNgModel]});
+
+                fixture = TestBed.createComponent(TestComponentWithNgModel);
+                fixture.detectChanges();
+                compiled = fixture.nativeElement;
+                dateContainerDebugElement = fixture.debugElement.query(By.directive(ClrDateContainer));
+                dateInputDebugElement = fixture.debugElement.query(By.directive(ClrDateInput));
+            });
+
+            it("accepts user input", fakeAsync(() => {
+                   fixture.componentInstance.dateValue = "01/02/2015";
+
+                   fixture.detectChanges();
+                   tick();
+
+                   expect(dateInputDebugElement.nativeElement.value).toBe("01/02/2015");
```

Comment:
Testing the native element here is just testing that `ngModel` works by itself. This exact test would work if you completely removed the `clrDate` directive from your test component.

You want to test that the service has the correct date, not the native element.

Reply:
fixed